### DISCUSSION TITLE
Import and track `#TODO:` items from the codebase

### DISCUSSION THREAD

**Discussion by dzintars** [GothenburgBitFactory/taskwarrior#3167] [Q&A]
I'm just exploring...
So... I have "general" Taskwarrior database, where I do put different stuff.
But lately i started to utilize `$TASKDATA` with [Direnv](https://direnv.net/) per project basis (software development). So instead of using Contexts or Projects I just use "local, per-project database". And I do commit it to Git (solo projects). That seems to work pretty well.

My question/idea is:

Is there some kind of Vim/Task plugin which can scan the entire project/code base for tags like `#TODO:` and to create the task list automatically for me? On top of that, I would love to have file path and line number of that particular `#TODO` item in annotation, so that when I click on it, it opens the new Vim buffer in that particular place.
???
Sometimes (quite often) it's more "ergonomic" just to slam `#TODO: Improve this thing` (example) comment just above the particular stanza and move on.
And when the comment is removed from the codebase, I just want it to be marked as Done.

Had anyone seen something similar like that?

**Comment by adriangalilea** [comment]
I would love this too

**Comment by adriangalilea** [comment]
@dzintars just so you know I managed to do something decent using both:

- [todo-to-issue-action](https://github.com/alstr/todo-to-issue-action)
- bugwarrior

I don't like the setup but I'll see how well it works for me over time.

You can also take a look at this:
https://github.com/ianlewis/todos
